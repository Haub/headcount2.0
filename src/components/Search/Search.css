.search {
  height: 2rem;
  width: 15rem;
}

.enter-button {
  height: 2rem;
  width: 7rem;
}