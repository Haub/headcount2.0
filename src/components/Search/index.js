import React, { Component } from 'react';
import './Search.css';
import PropTypes from 'prop-types';

class Search extends Component {

  handleChange = (event) => {
    this.props.filterDistricts(event.target.value);
  }

  render() {
    return (
      <form>
        <input
          placeholder="District name"
          onChange={this.handleChange}
          className='search'
        />
        <button className='enter-button'>Enter</button>
      </form>
    );
  }
}

Search.propTypes = { 
  filterDistricts: PropTypes.func
};

export default Search;