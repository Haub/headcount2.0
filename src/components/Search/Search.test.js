import React from 'react';
import { shallow } from 'enzyme';
import Search from './index';

describe('SEARCH', () => {
  
  it('should match the snapshot', () => {
    const wrapper = shallow(<Search />);
    expect(wrapper).toMatchSnapshot();
  });
  
  it('should invoke handleChange on the onChange of the input', () => {
    const handleChangeMock = jest.fn();
    const event = {target: { value: 'COLORADO'}};
    const wrapper = shallow(<Search filterDistricts={handleChangeMock}/>);
    wrapper.find('.search').simulate('change', event);
    expect(handleChangeMock).toHaveBeenCalled();
  });
});