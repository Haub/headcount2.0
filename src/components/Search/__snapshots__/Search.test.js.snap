// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SEARCH should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          className="search"
          onChange={[Function]}
          placeholder="District name"
/>,
        <button
          className="enter-button"
>
          Enter
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "search",
          "onChange": [Function],
          "placeholder": "District name",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Enter",
          "className": "enter-button",
        },
        "ref": null,
        "rendered": "Enter",
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="search"
            onChange={[Function]}
            placeholder="District name"
/>,
          <button
            className="enter-button"
>
            Enter
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "search",
            "onChange": [Function],
            "placeholder": "District name",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Enter",
            "className": "enter-button",
          },
          "ref": null,
          "rendered": "Enter",
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "disableLifecycleMethods": true,
    "hydrateIn": undefined,
  },
}
`;
